libpages/server.go

```diff
@@ -519,9 +493,13 @@ func ListenAndServe(ctx context.Context,
 		IdleTimeout:       httpIdleTimeout,
 	}
 
-	httpRedirectServer := http.Server{
-		Addr:              ":80",
-		Handler:           http.HandlerFunc(server.redirectHandlerFunc),
+	httpServer := http.Server{
+		Addr: ":80",
+		// Enable http-01 by calling the HTTPHandler method, and set the
+		// fallback HTTP handler to nil, which means we'll get a handler that
+		// redirects all HTTP traffic that's not for ACME domain verification
+		// to HTTPS.
+		Handler:           manager.HTTPHandler(nil),
```

Comment:
Does this need to be shutdown?

Reply:
It's a handler, so no. But the server that listens on `:80` is shutdown in the deferred function.